libfwupdplugin/fu-progress.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2009 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <gio/gio.h>
+
+#define FU_TYPE_PROGRESS (fu_progress_get_type())
+G_DECLARE_DERIVABLE_TYPE(FuProgress, fu_progress, FU, PROGRESS, GObject)
+
+struct _FuProgressClass {
+	GObjectClass parent_class;
+	/* signals */
+	void (*percentage_changed)(FuProgress *self, guint value);
```

Comment:
do you also want a signal for step changed?

Reply:
I don't _think_ so. If we needed one later we can easily add it.